x-pack/plugin/eql/src/main/antlr/EqlBase.g4

```diff
@@ -120,26 +125,16 @@ booleanValue
     : TRUE | FALSE
     ;
 
-qualifiedNames
-    : qualifiedName (COMMA qualifiedName)*
-    ;
-
 qualifiedName
```

Comment:
Why add the square brackets and integers? Not sure how that qualifies a name if there's no `.`.
The reason the grammar defines the prefix instead of the suffix optional is to help with parsing (the identifier is always the tail of the list while everything before it is the qualifier).

Reply:
This lets us type `a.b[0].c`, for instance. I believe we said indexing in arrays is out of scope, but it's sometimes necessary within pipes, because you can use `events[0].some_field` to refer to fields from the source events:

```sql
sequence by pid
  [process where process.name == "winword.exe"]
  [file where file.extension == "exe"]
| unique events[0].command_line
```

Reply:
I don't follow - if we don't support arrays how would that work with pipes?

As for array, grammar-wise the way to go is through a dedicated rule:
```
primaryExpression
  ...
 | qualifiedName 
 | value=primaryExpression LB index=valueExpression RB
```
This supports both literal expressions for the index and clearly differentiates between a name and the array syntax.

Reply:
This is more flexible than the existing EQL, which has its pros and cons.
This would accept this, which EQL doesn't currently accept:
```python
some_function(...)["some field"]
```

Reply:
Forgot to respond to this - to avoid having a function as an argument use qualifiedName or identifier instead in the array rule:
`| value=identifier LB index=valueExpression RB`

which would prevent the function rule from applying.